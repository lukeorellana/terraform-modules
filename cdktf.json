{
  "language": "typescript",
  "app": "npx ts-node",
  "terraformProviders": [],
  "projectId": "1b5f16bd-67ac-4af6-a33f-7bcafff141eb"
}