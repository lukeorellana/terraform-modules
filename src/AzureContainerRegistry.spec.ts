import { Testing, TerraformStack} from 'cdktf';
import {AzurermProvider} from "@cdktf/provider-azurerm";
import { AzureContainerRegistry } from './AzureContainerRegistry';
import 'cdktf/lib/testing/adapters/jest';


// test('renders a VPC with minimal config', () => {
//   const synthed = Testing.synthScope((stack) => {
//     new PocketVPC(stack, 'testPocketVPC');
//   });


describe('AzureContainerRegistry-Snapshot', () => {
  it('renders a AzureContainerRegistry without tags', () => {

    const synthed = Testing.synthScope((stack) => {
      new AzureContainerRegistry(stack, 'testACR', {
        name: 'acrtest888',
        location: 'eastus',
        resource_group_name: "rg-test",
        sku: "Premium",
        admin_enabled: false,
        georeplication_locations: ["EastUS"],
      });
    });
  
    expect(synthed).toMatchSnapshot();
  });

  it('renders a AzureContainerRegistry with tags', () => {
    const synthed = Testing.synthScope((stack) => {
      new AzureContainerRegistry(stack, 'testACR', {
        name: 'acrtest888',
        location: 'eastus',
        resource_group_name: "rg-test",
        sku: "Premium",
        admin_enabled: false,
        georeplication_locations: ["EastUS"],
        tags: {
          environment: "test",
        },
      });
    });
    expect(synthed).toMatchSnapshot();
  });
});


describe("AzureContainerRegistry-Terraform", () => {
  it("check if the produced terraform configuration is valid", () => {
    const app = Testing.app();
    const stack = new TerraformStack(app, "test");
    
    new AzurermProvider(stack, "azureFeature", {
      features: {},
    });


    new AzureContainerRegistry(stack, 'testACR', {
      name: 'acrtest888',
      location: 'eastus',
      resource_group_name: "rg-test",
      sku: "Premium",
      admin_enabled: false,
      georeplication_locations: ["EastUS"],
      tags: {
        environment: "test",
      },
    });


    // We need to do a full synth to validate the terraform configuration
    expect(Testing.fullSynth(stack)).toBeValidTerraform();
  });

  it("check if this can be planned", () => {
    const app = Testing.app();
    const stack = new TerraformStack(app, "test");

    new AzurermProvider(stack, "azureFeature", {
      features: {},
    });

    new AzureContainerRegistry(stack, 'testACR', {
      name: 'acrtest888',
      location: 'eastus',
      resource_group_name: "rg-test",
      sku: "Premium",
      admin_enabled: false,
      georeplication_locations: ["WestUS"],
      tags: {
        environment: "test",
      },
    });



    // We need to do a full synth to plan the terraform configuration
    expect(Testing.fullSynth(stack)).toPlanSuccessfully();
  });
});
