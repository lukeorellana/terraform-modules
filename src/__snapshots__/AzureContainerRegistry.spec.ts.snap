// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AzureContainerRegistry-Snapshot renders a AzureContainerRegistry and check snapshot 1`] = `
"{
}"
`;

exports[`AzureContainerRegistry-Snapshot renders a AzureContainerRegistry with tags 1`] = `
"{
  "output": {
    "id": {
      "value": "\${azurerm_container_registry.testACR_acr_53FDF08B.id}"
    }
  },
  "resource": {
    "azurerm_container_registry": {
      "testACR_acr_53FDF08B": {
        "admin_enabled": false,
        "georeplications": [
          {
            "location": "westus"
          }
        ],
        "location": "eastus",
        "name": "acrtest888",
        "resource_group_name": "rg-test",
        "sku": "Premium",
        "tags": {
          "environment": "test"
        }
      }
    }
  }
}"
`;

exports[`AzureContainerRegistry-Snapshot renders a AzureContainerRegistry without tags 1`] = `
"{
  "output": {
    "id": {
      "value": "\${azurerm_container_registry.testACR_acr_53FDF08B.id}"
    }
  },
  "resource": {
    "azurerm_container_registry": {
      "testACR_acr_53FDF08B": {
        "admin_enabled": false,
        "georeplications": [
          {
            "location": "westus"
          }
        ],
        "location": "eastus",
        "name": "acrtest888",
        "resource_group_name": "rg-test",
        "sku": "Premium"
      }
    }
  }
}"
`;
